exporters/autoexport/metrics.go

```diff
@@ -94,4 +100,53 @@ func init() {
 	RegisterMetricReader("none", func(ctx context.Context) (metric.Reader, error) {
 		return newNoopMetricReader(), nil
 	})
+	RegisterMetricReader("prometheus", func(ctx context.Context) (metric.Reader, error) {
+		// variable names and defaults specified at https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#prometheus-exporter
+		host := getenv("OTEL_EXPORTER_PROMETHEUS_HOST", "localhost")
+		port := getenv("OTEL_EXPORTER_PROMETHEUS_PORT", "9464")
+
+		mux := http.NewServeMux()
+		reg := prometheus.NewRegistry()
+		mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
+
+		// Timeouts are necessary to make a server resilent to attacks, but
+		// ListenAndServe doesn't set any. We use timeout values from this example: https://blog.cloudflare.com/exposing-go-on-the-internet/#:~:text=There%20are%20three%20main%20timeouts
+		// TODO: find defaults with a better justification, and/or make these configurable
+		go serve(ctx, &http.Server{
+			Addr:         host + ":" + port,
+			ReadTimeout:  5 * time.Second,
+			WriteTimeout: 10 * time.Second,
+			IdleTimeout:  120 * time.Second,
+			Handler:      mux,
+		})
+
+		return promexporter.New(promexporter.WithRegisterer(reg))
+	})
+}
+
+func serve(ctx context.Context, server *http.Server) {
+	serverErr := make(chan error, 1)
+	go func() {
+		serverErr <- server.ListenAndServe()
+	}()
+
+	var err error
+	select {
+	case <-ctx.Done():
+		err = server.Close()
+	case err = <-serverErr:
+	}
+
+	if err != nil && err != http.ErrServerClosed {
+		log.Printf("error closing Prometheus HTTP server: %v", err)
```

Comment:
AFAIK we should use [`otel.Handle`](https://pkg.go.dev/go.opentelemetry.io/otel#Handle) to report errors.

I suggest something like

`otel.Handle(fmt.Errorf("close Prometheus HTTP server: %w", err))`

Reply:
Done.